[docs] EAS Build: add docs about set-env (#23031)
expo/expo: docs/pages/build-reference/variables.mdx

See the [eas.json reference](/build/eas-json) for more information.

### Setting environment variables dynamically

You can also set environment variables dynamically during the build process. The `set-env` executable is available in the `PATH` on EAS Build workers, and can be used to set environment variables that will be visible in the next build phases.

For example, you can add the following in one of the [EAS Build hooks](/build-reference/npm-hooks/) and the environment variable `EXAMPLE_ENV` will be available until the end of the build job.

> ```bash
> set-env EXAMPLE_ENV "example value"

Reviewer: ```suggestion
<Terminal cmd={[ 'set-env EXAMPLE_ENV "example value"']} />
```

Let's use the `Terminal` component instead of a `bash` code snippet to represent the action to be done in the terminal or represent a command.

See the [eas.json reference](/build/eas-json) for more information.

### Setting environment variables dynamically

> You can also set environment variables dynamically during the build process. Executable `set-env` is available in the PATH on EAS Build worker, and can be used to set environment variables that will be visible in the next build phases.

Reviewer 2: ```suggestion
You can also set environment variables dynamically during the build process. The `set-env` executable is available in the `PATH` on EAS Build workers, and can be used to set environment variables that will be visible in the next build phases.
```

> For example, you can add
> ```bash
> set-env EXAMPLE_ENV "example value"
> ```
> in one of the EAS Build [hooks](/build-reference/npm-hooks/) and environment variable `EXAMPLE_ENV` will be available until the end of the build.

Reviewer 2: ```suggestion
For example, you can add the following in one of the [EAS Build hooks](/build-reference/npm-hooks/) and the environment variable `EXAMPLE_ENV` will be available until the end of the build job.

```bash
set-env EXAMPLE_ENV "example value"
```
```
